src/service/resource.js

```diff
@@ -413,13 +430,23 @@ export class Resource {
     return this.element.getUpgradeDelayMs();
   }
 
+  /**
+   * Stores a client rect that was "premeasured" by an IntersectionObserver.
+   * Should only be used in IntersectionObserver mode.
+   * @param {!ClientRect} clientRect
+   */
+  premeasure(clientRect) {
+    devAssert(this.intersect_);
+    this.premeasuredRect_ = clientRect;
+  }
+
   /**
    * Measures the resource's boundaries. An upgraded element will be
    * transitioned to the "ready for layout" state.
-   * @param {!ClientRect=} opt_premeasuredRect If provided, use this
-   *    premeasured ClientRect instead of calling getBoundingClientRect.
+   * @param {boolean} usePremeasuredRect If true, consumes the previously
+   *    premeasured ClientRect instead of calling getBoundingClientRect().
    */
-  measure(opt_premeasuredRect) {
+  measure(usePremeasuredRect = false) {
```

Comment:
Would it make sense to always use `premeasuredRect` if available? When will there ever be a premeasured rect that we don't intend on using?

Reply:
> Yes, sometimes we still do explicit measures e.g. after a component requests a size-change.

In that case, there shouldn't be a `premeasuredRect` though, since it is set to null